### ISSUE TITLE
My work app won't pick up my location

### ISSUE THREAD

**Issue by haros3000** [GrapheneOS/os-issue-tracker#2474]
I have a Pixel 7 running latest software. This app works flawlessly on stock Android but I can't get it to detect my location on Graphene https://play.google.com/store/apps/details?id=com.app.hubfleet&hl=en_AU&gl=US

The app isn't freezing, everything else in it works fine but I need it to detect my location which it won't do. Other apps which require my location are able to detect it in Graphene eg. Organic Maps and Magic Earth both work fine.

The app has permission to access location and I have toggled all settings in this configuration guide, plus a few others, with no luck https://grapheneos.org/usage#sandboxed-google-play-configuration

I gave Google Play services the exact same permissions my stock Android phone (where the app works) has but still doesn't work.

The GPSTest app by barbeauDev confirms I am getting a lock.

Someone suggested I create a fresh profile, install sandboxed Google Play, install this app and give it location permission without changing any other settings. Still won't pick up my location.

**Comment by thestinger** [comment]
It sounds like you're disabling location request re-routing and using the Google Play location service. Did you enable the network accuracy toggle for it?

It's possible the app only tries to use network location via the OS. We only provided emulated network location via GNSS through the OS location service, which should still work especially if you're using GPSTest before trying this app.

**Comment by haros3000** [comment]
Yes, I enabled precise location and have also tried enabling location rerouting through the OS. Tried allowing unrestricted battery usage for Play services. Still nothing.

Also notified the app developer. He said he'll try to fix it in the next release but I don't think he knows what the issue is or whether he can fix it yet.

**Comment by simonwiles** [comment]
The `Merlin Bird ID` (available on the play store) seems to have the same issue -- I haven't found any other apps that are unable to get my location (though I don't use many different apps tbh), but no combination of location-/permission-related settings seem to work with this app.  Have confirmed the app works fine on stock.  Following here in case of new information :)

**Comment by thestinger** [comment]
Please update to the latest release. It may be fixed.

**Comment by haros3000** [comment]
> The `Merlin Bird ID` app (available on the play store) seems to have the same issue -- I haven't found any other apps that are unable to get my location (though I don't use many different apps tbh), but no combination of location-/permission-related settings seem to work with this app. Have confirmed the app works fine on stock. Following here in case of new information :)

Can you update to the latest release and confirm whether it's fixed? I already sold my Pixel as I need that app daily, but I'll buy another if it's working

**Comment by simonwiles** [comment]
I have the new release and the new options available, but no change, I'm afraid; all the apps that worked before still work, but the `Merlin Bird ID` app still does not, no matter what options I select.  It's listed as requesting and being granted location permission, but it never works.  In some contexts the app reports "service unavailable", but that's an app-side message so it's not terribly helpful.

**Comment by haros3000** [comment]
I had also tried toggling every setting that could be remotely related including bluetooth/wifi scanning, permission for nearby devices, location accuracy etc.

The sandboxed google play config guide says "The Google Location Accuracy and Google settings activities would normally be integrated into the OS but we don't include any of the standard Google Play integration so there needs to be an app providing a way to access them."

Could the apps be missing this or perhaps have an attestation compatibility issue? Either of these would mean we need to ask the app developer to make appropriate changes, but they're more likely to fix it if we could help narrow it down for them
